docs/code-guidelines.md

```diff
@@ -0,0 +1,1141 @@
+
+Consider what it will look like in godoc. Named result parameters like:
+
+```go
+func (n *Node) Parent1() (node *Node) {}
+func (n *Node) Parent2() (node *Node, err error) {}
+```
+
+It will be repetitive in godoc; better to use:
+
+```go
+func (n *Node) Parent1() *Node {}
+func (n *Node) Parent2() (*Node, error) {}
+```
+
+On the other hand, adding names may be helpful in some contexts if a function returns two or three parameters of the same type or if the meaning of a result isn't clear from the context. Don't name result parameters just to avoid declaring a var inside the function; that trades off minor implementation brevity at the cost of unnecessary API verbosity.
+
+```go
+func (f *Foo) Location() (float64, float64, error)
+```
+
+It is less clear than the:
+
+```go
+// Location returns f's latitude and longitude.
+// Negative values mean south and west, respectively.
+func (f *Foo) Location() (lat, long float64, err error)
+```
+
+Naked returns are okay if the function is a handful of lines. Once it's a medium-sized function, be explicit with your return values. Corollary: it's not worth naming result parameters just because it enables you to use naked returns. Clarifying docs is always more important than saving a line or two in your function.
+
+Finally, it would help if you named a result parameter in some cases to change it in a deferred closure. That is always OK.
+
+
+### Package Comments
+
+Package comments, like all comments to be presented by godoc, must appear adjacent to the package clause with no blank line.
+
+```go
+/*
+Package template implements data-driven templates for generating textual
+output such as HTML.
+....
+*/
+package template
+```
+
+For "package main" comments, other styles of comment are fine after the binary name (and it may be capitalized if it comes first). For example, for a `package main` in the directory `seedgen` you could write:
+
+```go
+// Seedgen ..
+package main
+```
+
+See https://go.dev/doc/effective_go#commentary for more information about commentary conventions.
+
+### Package Names
+
+All references to names in your package will be done using the package name so that you can omit that name from the identifiers. For example, if you are in package chubby, you don't need to type ChubbyFile, which clients will write as `chubby.ChubbyFile`. Instead, name the type `File`, which clients will write as `chubby.File`. Avoid meaningless package names like util, common, misc, API, types, and interfaces. See https://go.dev/doc/effective_go#package-names and https://go.dev/blog/package-names for more.
+
+When naming packages, choose a name that is:
+
+- All lowercase. No capitals or underscores.
+- Does not need to be renamed using named imports at most call sites.
+- Short and succinct. Remember that the name is identified in full at every call site.
+- Not plural. For example, `net/url`, not `net/urls`.
+- Not "common", "util", "shared", or "lib". These are bad, uninformative names.
+
```

Comment:
to distinguish sdk and gaia with same package name, add `sdk` or `gaia` or `modulename` as prefix.
for example: `sdk/types`, `gaia/types` and `gaia/x/globalfee/types`, can use `sdktype`, `gaiatype`, `globalfeetype`

Reply:
nice!!! I will add it